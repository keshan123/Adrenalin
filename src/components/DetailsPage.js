import React, { Component } from 'react';
import { Switch } from 'react-router-dom'

import {
  PageContainer,
  DetailsContainer,
  DetailsImage,
  DetailsInfo,
  DetailsTitle,
  DetailsQuestionsContainer,
  DetailsQuestionsTitle,
  DetailsQuestionsBody,
  DetailsTagContainer,
  DetailsImageTag,
} from '../styles';

let data = require('../feed/data.json');

class DetailsPage extends Component {
  state = {
    title: '',
    questions: [],
    tag: '',
    image: '',
  }
  componentDidMount() {
    const isFound = (casestudy) => {
      if (casestudy.id.toString() === this.props.match.params.id) {
        this.setState({
          title: casestudy.title,
          questions: casestudy.questions,
          tag: casestudy.tag,
          image: casestudy.image,
        })
      }
    }
    data.find(isFound)
  }
  render() {
    return (
      <Switch>
        <PageContainer>
          <DetailsContainer>
            {
              this.state.image &&
              <DetailsTagContainer>
                <DetailsImageTag>
                 {this.state.tag}
                </DetailsImageTag>
                <DetailsImage
                  image={require(`../assets/${this.state.image}`)}
                />
              </DetailsTagContainer>
            }
            <DetailsInfo>
              <DetailsTitle>
                {this.state.title}
              </DetailsTitle>
              {
                this.state.questions.map((caseStudy, index) =>
                <DetailsQuestionsContainer key={index}>
                  <DetailsQuestionsTitle>
                    Question {index + 1}
                  </DetailsQuestionsTitle>
                  <DetailsQuestionsBody>
                    {caseStudy}
                  </DetailsQuestionsBody>
                </DetailsQuestionsContainer>
                )
              }
            </DetailsInfo>
          </DetailsContainer>
        </PageContainer>
      </Switch>
    );
  }
}

export default DetailsPage;
