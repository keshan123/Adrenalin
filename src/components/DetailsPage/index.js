import React, { Component } from 'react';
import { Switch } from 'react-router-dom'

import {
  PageContainer,
  DetailsContainer,
  DetailsImage,
  DetailsInfo,
  DetailsTitle,
  DetailsQuestionsContainer,
  DetailsQuestionsTitle,
  DetailsQuestionsBody,
  DetailsTagContainer,
  DetailsImageTag,
} from './styles';

let data = require('../../feed/data.json');

class DetailsPage extends Component {
  state = {
    title: '',
    questions: [],
    tag: '',
    image: '',
  };

  componentDidMount() {
    const isCaseStudyFound = (casestudy) => {
      if (casestudy.id.toString() === this.props.match.params.id) {
        this.setState({
          title: casestudy.title,
          questions: casestudy.questions,
          tag: casestudy.tag,
          image: casestudy.image,
        })
      }
    }
    data.find(isCaseStudyFound)
  };

  render() {
    const { image, title, questions } = this.state;
    return (
      <Switch>
        <PageContainer>
          <DetailsContainer>
            {
              image &&
              <DetailsTagContainer>
                <DetailsImageTag>
                 {this.state.tag}
                </DetailsImageTag>
                <DetailsImage
                  image={require(`../../assets/${image}`)}
                />
              </DetailsTagContainer>
            }
            <DetailsInfo>
              <DetailsTitle>
                {title}
              </DetailsTitle>
              {
                questions.map((caseStudy, index) =>
                <DetailsQuestionsContainer key={index}>
                  <DetailsQuestionsTitle>
                    Question {index + 1}
                  </DetailsQuestionsTitle>
                  <DetailsQuestionsBody>
                    {caseStudy}
                  </DetailsQuestionsBody>
                </DetailsQuestionsContainer>
                )
              }
            </DetailsInfo>
          </DetailsContainer>
        </PageContainer>
      </Switch>
    );
  };
}

export default DetailsPage;
