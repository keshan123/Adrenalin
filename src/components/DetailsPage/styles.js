import styled from 'styled-components';

// DetailsPage

export const PageContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const DetailsContainer = styled.div`
  align-items: flex-start;
  display: flex;
  justify-content: space-between;
  padding: 0 93px;

  @media (max-width: 830px) {
    padding: 0;
    align-items: center;
    flex-direction: column;
  }
`;

export const DetailsImage = styled.div`
  background-image: url(${props => props.image});
  background-size: cover;
  height: 1107px;
  margin-top: 62px;
  width: 650px;
`;

export const DetailsInfo = styled.div`
  height: 800px;
  margin-left: 80px;
  margin-top: 123px;
  overflow: scroll;
  width: 564px;

  @media (max-width: 830px) {
    margin-left: 0;
  }
`;

export const DetailsTitle = styled.div`
  align-text: left;
  font-family: Arial,Helvetica Neue,Helvetica,sans-serif;
  font-size: 68px;
  font-weight: bold;
  margin-bottom: 65px;
`;

export const DetailsQuestionsContainer = styled.div`
  align-text: left;
  margin-bottom: 30px;
`;

export const DetailsQuestionsTitle = styled.div`
  font-size: 32px;
  font-weight: bold;
`;

export const DetailsQuestionsBody = styled.div`
  color: #333333;
  font-family: Georgia,Times,Times New Roman,serif;
  font-family: Georgia;
  font-size: 21px;
  font-weight: normal;
  letter-spacing: 0;
  line-height: 37px;
  margin-top: 10px;
`;

export const DetailsTagContainer = styled.div`
  position: relative;
`;

export const DetailsImageTag = styled.div`
  align-items: flex-start;
  background-color: #F5F5F5;
  box-shadow: 0px 4px 4px -2px rgba(0,0,0,0.3);
  color: rgba(0,0,0,0.75);
  display: flex;
  font-size: 12px;
  height: 24px;
  justify-content: center;
  left: 0;
  margin-left: -37px;
  margin-top: 99px;
  position: absolute;
  top: 0;
  transform: rotate(270deg);
  width: 98px;
`;
