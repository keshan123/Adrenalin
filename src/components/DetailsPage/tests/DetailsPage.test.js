import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom';
import 'jest-styled-components'
import DetailsPage from '../index'

const match = { params : { id : 1  } }

test('DetailsPage renders', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <DetailsPage
        match={match}
      />
    </BrowserRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
