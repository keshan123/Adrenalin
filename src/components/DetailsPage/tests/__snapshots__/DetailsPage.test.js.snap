// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 93px;
}

.c2 {
  height: 800px;
  margin-left: 80px;
  margin-top: 123px;
  overflow: scroll;
  width: 564px;
}

.c3 {
  -webkit-align-text: left;
  -ms-flex-line-packalign-text: left;
  align-text: left;
  font-family: Arial,Helvetica Neue,Helvetica,sans-serif;
  font-size: 68px;
  font-weight: bold;
  margin-bottom: 65px;
}

@media (max-width:830px) {
  .c1 {
    padding: 0;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:830px) {
  .c2 {
    margin-left: 0;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        
      </div>
    </div>
  </div>
</div>
`;
